test/FunctionalTests/Microsoft.OData.Edm.Tests/Csdl/CsdlReaderTests.TargetPath.cs

```diff
@@ -0,0 +1,465 @@
+        }
+
+        [Fact]
+        public void ParsingInvalidAnnotationTargetValidationFails()
+        {
+            string types =
+@"
+<Annotations Target=""NS.Default/Me/Orders/NS.VipOrder/OrderId"" >
+  <Annotation Term=""NS.MyTerm"" String=""Name OutOfLine MyTerm Value"" />
+</Annotations>";
+
+            string properties ="";
+
+            IEdmModel model = GetModel(types: types, properties: properties);
+            Assert.NotNull(model);
+
+            IEnumerable<EdmError> errors;
+            Assert.False(model.Validate(out errors));
+            var error1 = errors.First();
+            var error2 = errors.ElementAt(1);
+            Assert.Equal(EdmErrorCode.BadUnresolvedProperty, error1.ErrorCode);
+            Assert.Equal(EdmErrorCode.BadUnresolvedTarget, error2.ErrorCode);
+        }
+
+        [Fact]
+        public void ParsingMultipleOutOfLineAnnotationsWorks()
+        {
+            string types =
+@"<EntityType Name=""Order"">
+    <Key>
+        <PropertyRef Name=""OrderId""/>
+    </Key>
+    <Property Name=""OrderId"" Type=""Edm.String"" Nullable=""false""/>
+    <Property Name=""OrderName"" Type=""Edm.String"" Nullable=""false""/>
+</EntityType>
+<Term Name=""MyTerm"" Type=""Edm.String""/>
+<Term Name=""MyTerm2"" Type=""Edm.String""/>
+<Annotations Target=""NS.Default/Me/Orders"" >
+  <Annotation Term=""NS.MyTerm"" String=""Name OutOfLine MyTerm Value"" />
+</Annotations>
+<Annotations Target=""NS.Default/Me/Orders"" >
+  <Annotation Term=""NS.MyTerm2"" String=""Name OutOfLine MyTerm Value 2"" />
+</Annotations>
+<Annotations Target=""NS.Default/Customers/Orders"" >
+  <Annotation Term=""NS.MyTerm"" String=""Name OutOfLine MyTerm Value"" />
+</Annotations>";
+
+            string properties =
+                @"<NavigationProperty Name=""Orders"" Type=""Collection(NS.Order)"" ContainsTarget=""true"" />";
+
+            IEdmModel model = GetModel(types: types, properties: properties);
+            Assert.NotNull(model);
+
+            IEnumerable<EdmError> errors;
+            Assert.True(model.Validate(out errors), String.Format("Errors in validating model. {0}", String.Concat(errors.Select(e => e.ErrorMessage))));
+
+            var customer = model.SchemaElements.OfType<IEdmEntityType>().FirstOrDefault(c => c.Name == "Customer");
+            Assert.NotNull(customer);
+
+            IEdmTerm myTerm = model.FindDeclaredTerm("NS.MyTerm");
+            Assert.NotNull(myTerm);
+
+            // Name
+            IEdmProperty nameProperty = customer.DeclaredNavigationProperties().Where(x => x.Name == "Orders").FirstOrDefault();
+            Assert.NotNull(nameProperty);
+
+            // Mark model as Immutable so as to cache annotations.
+            model.MarkAsImmutable();
+
+            IEdmTargetPath targetPath = model.GetTargetPath("NS.Default/Me/Orders");
+
+            string nameAnnotation = GetAnnotation(model, targetPath, myTerm, EdmVocabularyAnnotationSerializationLocation.OutOfLine);

```

Comment:
This should return 2 annotations since the targetpath has 2 annotations

Reply:
We are returning the annotation with the specified term